libvast/src/system/connector.cpp

```diff
@@ -0,0 +1,168 @@
+    VAST_WARN("unable to retrieve error code for a remote node connection "
+              "error:{}",
+              err);
+    return false;
+  }
+  return is_non_recoverable_error_enum(err_enum);
+}
+} // namespace
+
+connector_actor::behavior_type
+connector(connector_actor::stateful_pointer<connector_state> self,
+          std::size_t max_connection_attempts,
+          caf::timespan connection_retry_delay, vast::port port,
+          std::string host, caf::timespan connection_timeout) {
+  self->state.connection_retry_delay = connection_retry_delay;
+  self->state.port = std::move(port);
+  self->state.host = std::move(host);
+  self->state.node_connection_timeout = connection_timeout;
+  return {[self,
+           max_connection_attempts](atom::connect) -> caf::result<node_actor> {
+            if (self->state.promise.pending())
+              return self->state.promise;
+            self->state.remaining_connection_attempts = max_connection_attempts;
+            self->state.promise = self->make_response_promise<node_actor>();
+            VAST_INFO("client connects to VAST node at {}", self->state.host);
+            self->send(self, atom::internal_v, atom::start_v);
+            return self->state.promise;
+          },
+          [self](atom::internal, atom::start) {
+            auto middleman = self->system().has_openssl_manager()
+                               ? self->system().openssl_manager().actor_handle()
+                               : self->system().middleman().actor_handle();
+            self
+              ->request(middleman, self->state.node_connection_timeout,
+                        caf::connect_atom_v, self->state.host,
+                        self->state.port.number())
+              .then(
+                [self](const caf::node_id& node_id, caf::strong_actor_ptr& node,
+                       const std::set<std::string>&) {
+                  VAST_DEBUG("client connected to VAST node: {} at {}:{}",
+                             to_string(node_id), self->state.host,
+                             to_string(self->state.port));
+                  self->state.promise.deliver(
+                    caf::actor_cast<node_actor>(std::move(node)));
+                },
+                [self](caf::error& err) {
+                  if (is_non_recoverable_error(err)
+                      || --self->state.remaining_connection_attempts == 0) {
+                    self->state.promise.deliver(caf::make_error(
+                      ec::system_error,
+                      fmt::format("failed to connect to VAST node at {}:{}: {}",
+                                  self->state.host, self->state.port.number(),
+                                  std::move(err))));
+                  } else {
+                    self->delayed_send(self, self->state.connection_retry_delay,
+                                       atom::internal_v, atom::start_v);
+                  }
+                });
+          }};
```

Comment:
```suggestion
          },};
```

And then reformat to get rid of a few levels of unnecessary indentation.